File: RedSheet.DbEntities/Models/vOurRequirementDetailRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vOurRequirementDetailRecords", Schema = "dbo")]
	[Serializable]
	public partial class vOurRequirementDetailRecord  
	{
 
	[MaxLength(1000)]
	public string FourStep { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string ldo { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string mdo { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int OurRequirementDetailId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int PainFactorId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectRequirementId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Requirement { get; set; } 
 
	[MaxLength(1000)]
	public string SecondStep { get; set; } 
 
	[MaxLength(1000)]
	public string ThirdStep { get; set; } 

	public vOurRequirementDetailRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-our-requirement-detail-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vOurRequirementDetailRecord {
    constructor(vOurRequirementDetailRecord?: vOurRequirementDetailRecord )  {
        let properties = [ "fourStep", "ldo", "mdo", "ourRequirementDetailId", "painFactorId", "projectRequirementId", "requirement", "secondStep", "thirdStep",];
        for (let property of properties)
            if (vOurRequirementDetailRecord && vOurRequirementDetailRecord[property])
                this[property] = vOurRequirementDetailRecord[property];
    }
 
    @maxLength(1000)
	fourStep : string =   undefined;
 
    @required()
    @maxLength(1000)
	ldo : string =   undefined;
 
    @required()
    @maxLength(1000)
	mdo : string =   undefined;
 
	ourRequirementDetailId : number =   0 ;
 
    @range(1,2147483647)
	painFactorId : number =   undefined;
 
    @range(1,2147483647)
	projectRequirementId : number =   undefined;
 
    @required()
    @maxLength(1000)
	requirement : string =   undefined;
 
    @maxLength(1000)
	secondStep : string =   undefined;
 
    @maxLength(1000)
	thirdStep : string =   undefined;


}
